internal/services/cockpit/datasource.go

```diff
@@ -0,0 +1,147 @@
+			Default: schema.DefaultTimeout(DefaultCockpitTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Name of the datasource",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The type of the datasource",
+				ValidateFunc: validation.StringInSlice([]string{
+					cockpit.DataSourceTypeMetrics.String(),
+					cockpit.DataSourceTypeLogs.String(),
+					cockpit.DataSourceTypeTraces.String(),
+				}, false),
+			},
+			// computed
+			"url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL of the datasource",
+			},
+			"origin": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The origin of the datasource",
+			},
+			"synchronized_with_grafana": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Indicates whether the data source is synchronized with Grafana",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the creation of the cockpit datasource",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the cockpit datasource",
+			},
+			"project_id": account.ProjectIDSchema(),
+			"region":     regional.Schema(),
+		},
+	}
+}
+
+func ResourceCockpitDatasourceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := cockpitAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	res, err := api.CreateDataSource(&cockpit.RegionalAPICreateDataSourceRequest{
+		Region:    region,
+		ProjectID: d.Get("project_id").(string),
+		Name:      d.Get("name").(string),
+		Type:      cockpit.DataSourceType(d.Get("type").(string)),
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(regional.NewIDString(region, res.ID))
+	return ResourceCockpitDatasourceRead(ctx, d, meta)
+}
+
+func ResourceCockpitDatasourceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, id, err := NewAPIWithRegionAndID(meta, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	res, err := api.GetDataSource(&cockpit.RegionalAPIGetDataSourceRequest{
+		Region:       region,
+		DataSourceID: id,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		if httperrors.Is404(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	_ = d.Set("name", res.Name)
+	_ = d.Set("type", res.Type.String())
+	_ = d.Set("url", res.URL)
+	_ = d.Set("origin", res.Origin)
+	_ = d.Set("synchronized_with_grafana", res.SynchronizedWithGrafana)
```

Comment:
Should it contains a waiter for `synchronized_with_grafana` ? Is this required to use the datasource ?

Reply:
We don't have to wait for this field for the resource to be ready